src/angular/components/add-edit.component.ts

```diff
@@ -221,6 +223,10 @@ export class AddEditComponent implements OnInit {
     }
 
     async submit(): Promise<boolean> {
+        if (this.cipher.isDeleted) {
+            return this.restore();
+        }
```

Comment:
I guess this is just a safety check? This shouldn't ever happen right?

Reply:
No, this was when the user clicks on a cipher from the list in their vault, it opens the add-edit component modal, which only has a single button - Submit (labelled "Save") which I relabel "Restore" as the only viable option, and there's only one handler for onSubmit -> submit() so I'm hijacking that if the cipher is deleted to simply redirect to the restore action. I'm open to making 2 buttons on the html template with `*ngIf` or some other direction and different event handler if that appears like a cleaner implementation; Open to either approach.